### PR TITLE
Feature/docus add doxygen for code addgrid.f

### LINKED ISSUE #59: doxyfile needs to be updated and some settings turned on
Same as https://github.com/NOAA-EMC/NCEPLIBS-w3emc/issues/81.

There are only about 65 files, so if we can do 5 a day, we can have everything done in 3 weeks. I did one today and it took about 10 minutes.

At the suggestion of @arunchawla-NOAA this should be done before the documentation for w3emc.

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
Add doxygen one by one.

### CODE REVIEW

File: src/addgrid.f [hunk 1]
```diff
@@ -1,67 +1,47 @@
+!>    @file
+!>    @brief This subroutine packs up a Grid Definition Section
+!>    (Section 3) and adds it to a GRIB2 message.
+!>    @author Gilbert ORG: W/NP11 @date 2000-05-01
+!>
```
Comment (edwardhartnett) [inline src/addgrid.f]: Need to doocumentation blocks.

Basically, end the first one, then a blank line, then start a new one.

File: src/addgrid.f [hunk 2]
```diff
@@ -1,67 +1,47 @@
...
+!>    for which optional grid points are defined.
+!>    @param[out] ierr Error return code.
+!>    - 0 = no error
+!>    - 1 = GRIB message was not initialized.  Need to call routine gribcreate first.
+!>    - 2 = GRIB message already complete.  Cannot add new section.
+!>    - 3 = Sum of Section byte counts doesn't add to total byte count.
+!>    - 4 = Previous Section was not 1, 2 or 7.
+!>    - 5 = Could not find requested Grid Definition Template.
+!>
+!>    REMARKS: Note that the Local Use Section ( Section 2 ) can only follow
```
Comment (edwardhartnett) [inline src/addgrid.f]: Change REMARKS to use @note tag instead:

`@note The Local Use Section ( Section 2 ) can only follow...`

File: src/addgrid.f [hunk 3]
```diff
@@ -1,67 +1,47 @@
...
+!>          Section 1 or Section 7 in a GRIB2 message.
+!>
```
Comment (edwardhartnett) [inline src/addgrid.f:41]: Subroutine @author and @date tag needed here. Just a repeat of the one used in the file doxygen block.

**Review by edwardhartnett — APPROVED** [review]
Much closer!

**Comment by edwardhartnett** [comment]
Also please change the PR title to include the name of the file you are fixing.

Just using the branch name as a PR title is not very helpful - we need more detail, so we can easily tell what each PR is doing.

File: src/addgrid.f (hunk 1 above)
Comment (Hang-Lei-NOAA) [reply src/addgrid.f]: addressed, Thanks!

Comment (Hang-Lei-NOAA) [reply src/addgrid.f]: addressed, Thanks!

File: src/addgrid.f (hunk 2 above)
Comment (Hang-Lei-NOAA) [reply src/addgrid.f]: addressed, Thanks!

File: src/addgrid.f (hunk 3 above)
Comment (Hang-Lei-NOAA) [reply src/addgrid.f:41]: add it, Thanks!

**Pull request merged**